### DISCUSSION TITLE
[Feature Request] New Commands: Text Link using {{clipboard}} or {{selection}}

### DISCUSSION THREAD

**Discussion by FelipeRearden** [jglev/obsidian-carry-forward#5] [Ideas]
Hello @jglev !!!!!

I'd like to share with you one workflow that I think could be a great feature for Carry-Foward. 

## In-line embeds (link text)

My idea is to automate the workflow of using phrases (text) inside the alias component of a blockquote (link text).

### For example: 

Note A 

This is my text about a beautiful day

-> **I want to create a block ref using the selected/copied text from Note A on Note B like this:**

Note B

All I want is a good life and [[Note A#^qwe123 | This is my text about a beautiful day]]

I'm calling this type of block ref In-line Embeds.

## My FR

New commands to use {{selection}} and/or {{clipboard}} as link text 

Carry Forward: Copy Link to line using Clipboard
Carry Forward: Copy Link to line using Selection

In my example I would select or copy to the clipboard the phrase `This is my text about a beautiful day` than hit one of the new commands 
- Than go to Note B and paste.

---

Thanks for reading this :)

Have a great day!!!!!

**Comment by jglev** [comment]
Thank you for writing this up, @FelipeRearden! This is now addressed (I hope!) in `v1.3.0` : )

**Reply by FelipeRearden** [reply]
> Thank you for writing this up, @FelipeRearden! This is now addressed (I hope!) in `v1.3.0` : )

You have no idea how happy I'm @jglev :)

Thank you very much for making my FR dream a reality. YOU ARE SUPER :)

---

Question for @jglev 

There is a post on forum related to this FR. As far as I know, the new version 1.3.0 is exactly what our friends from the forum want:

https://forum.obsidian.md/t/a-proposal-for-rendering-block-references-inline/27093

I wanna ask you if I can share your plugin and your git link from Carry-Forward in this post on the forum.

I don't wanna do something before ask your permission :)

**Reply by jglev** [reply]
Thank you for your kind words! : D

> I wanna ask you if I can share your plugin and your git link from Carry-Forward in this post on the forum.

Certainly! Please feel free to share it wherever you like! (And while I very much appreciate your deference, no need to ask! It's a public repo, and I'm happy for others to know about it.)

**Reply by jglev** [reply]
I'm also not active on the forum, so I'm doubly happy for you doing so : )

**Reply by FelipeRearden** [reply]
> Certainly! Please feel free to share it wherever you like! (And while I very much appreciate your deference, no need to ask! It's a public repo, and I'm happy for others to know about it.)

😃 I will, I really hope that people like this new feature as much as I do :)

> I'm also not active on the forum, so I'm doubly happy for you doing so : )

Yeah! Before I create this FR I search on Discord and forum for some knowledge. When I saw that many people are looking for this, my mind just EUREKA and saw that this is something that only Carry-Forward could do. 

Thanks for being such a great person @jglev :)
